modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/ServerApiAccess.java

```diff
@@ -0,0 +1,327 @@
+		String httpMethod = apiUtilities.getHttpMethod();
+		String url = apiUtilities.getEndPoint();
+		Map<String, String> params = apiUtilities.getRequestParamsMap();
+
+		Map<String, String> response_params = new HashMap<String, String>();
+		HttpClient httpclient = getCertifiedHttpClient();
+		String payload = buildPayload(params);
+
+		if (httpMethod.equals("POST")) {
+			HttpPost httpPost = new HttpPost(url);
+			Log.e("url", "" + url);
+			HttpPost httpPostWithHeaders = (HttpPost) buildHeaders(httpPost, headers, httpMethod);
+			byte[] postData = payload.getBytes();
+			try {
+				httpPostWithHeaders.setEntity(new ByteArrayEntity(postData));
+				HttpResponse response = httpclient.execute(httpPostWithHeaders);
+				String status = String.valueOf(response.getStatusLine().getStatusCode());
+				Log.d(TAG, status);
+				response_params.put("response", getResponseBody(response));
+				response_params.put("status", status);
+				return response_params;
+			} catch (ClientProtocolException e) {
+				Log.d(TAG, "ClientProtocolException :" + e.toString());
+				return null;
+			} catch (IOException e) {
+				Log.d(TAG, e.toString());
+				response_params.put("response", "Internal Server Error");
+				response_params.put("status", "500");
+				return response_params;
+			}
+		}
+		return null;
+	}
+
+	public static HttpClient getCertifiedHttpClient() {
+		try {
+			HttpClient client = null;
+			if (Constants.SERVER_PROTOCOL.equalsIgnoreCase("https://")) {
+				KeyStore localTrustStore = KeyStore.getInstance("BKS");
+				InputStream in =
+						IdentityProxy.getInstance().getContext().getResources()
+						             .openRawResource(R.raw.emm_truststore);
+				localTrustStore.load(in, Constants.TRUSTSTORE_PASSWORD.toCharArray());
+
+				SchemeRegistry schemeRegistry = new SchemeRegistry();
+				schemeRegistry.register(new Scheme("http", PlainSocketFactory.getSocketFactory(),
+				                                   80));
+				SSLSocketFactory sslSocketFactory = new SSLSocketFactory(localTrustStore);
+				sslSocketFactory.setHostnameVerifier(SSLSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
+				schemeRegistry.register(new Scheme("https", sslSocketFactory, 443));
+				HttpParams params = new BasicHttpParams();
+				ClientConnectionManager cm =
+						new ThreadSafeClientConnManager(params, schemeRegistry);
+
+				client = new DefaultHttpClient(cm, params);
+
+			} else {
+				client = new DefaultHttpClient();
+			}
+			return client;
+		} catch (Exception e) {
+			Log.d(TAG, e.toString());
+			return null;
+		}
+	}
+
+	public static String getResponseBody(HttpResponse response) {
+
+		String response_text = null;
+		HttpEntity entity = null;
+		try {
+			entity = response.getEntity();
+			response_text = getResponseBodyContent(entity);
+		} catch (ParseException e) {
+			Log.d(TAG, e.toString());
+		} catch (IOException e) {
+			if (entity != null) {
+				try {
+					entity.consumeContent();
```

Comment:
Shouldn't this be handled in a finally block? In other words, shouldn't the entity be consumed irrespective of whether the method call is a success or a failure?